### ISSUE TITLE
NodeJS require paths are relative to the proto/grpc rule

### ISSUE THREAD

**Issue by mmatheson** [rules-proto-grpc/rules_proto_grpc#107]
The require path for the `_pb.js` files generated by the `nodejs_*` rules is inconsistent with other language implementations. 

In the [cpp example](https://github.com/rules-proto-grpc/rules_proto_grpc/blob/master/cpp/example/routeguide/server.cc#L32): 
`#include "example/proto/routeguide.grpc.pb.h"` matches the location of the routeguide proto in the repo (`example/proto/routeguide.proto`).

In the [nodejs example](https://github.com/rules-proto-grpc/rules_proto_grpc/blob/master/nodejs/example/routeguide/client.js#L4): `require('routeguide/routeguide_pb/example/proto/routeguide_pb.js')` appears to be `{path/to/nodejs_grpc_library}/example/proto/routeguide.proto`. This inconsistency leads to long, hard to parse require paths, and obscures the actual location of the .proto file when reading source code.

**Comment by aaliddell** [comment]
This is because the `cc_library` rule provides a mechanism for expressing the include root with the `includes` attribute, which allows us to hide the generated prefix. With `js_library`, we are only able to provide the package name, which in this case is `routeguide`. The path is then generated statically for any srcs: https://github.com/bazelbuild/rules_nodejs/blob/2c2cc6eec42b0f4f805fb3063920107b78f821fd/internal/js_library/js_library.bzl#L226

To fix this, we'd either need to have support from rules_nodejs for producing a package that has a path pointing into the `routeguide_pb` directory (similar to prefix stripping in other rules) or otherwise wrap/patch the produced `LinkablePackageInfo` provider to amend the path.

In either case, the import would become `require('routeguide/example/proto/routeguide_pb.js')`, not just `require('example/proto/routeguide_pb.js')`, since a package name is needed regardless.

**Comment by aaliddell** [comment]
Pending on https://github.com/bazelbuild/rules_nodejs/issues/2482

**Comment by andrei-abramov** [comment]
Experiencing the same issue with `js_grpc_web_compile` - the only workaround I found is to hack the `prefix_path` attribute to point to the workspace root e.g.

```
js_grpc_web_compile(
    name = "echo_proto_web",
    protos = [":echo_proto"],
    visibility = ["//visibility:public"],
    prefix_path = "../../.."
)
```

**Comment by aaliddell** [comment]
There will hopefully be a fix for this coming: https://github.com/bazelbuild/rules_nodejs/pull/2519

To maintain backwards compatibility, this'll have to be an opt-in mode at least for the 3.x.x cycle.

**Comment by aaliddell** [comment]
Support for the required prefix stripping feature in rules_nodejs landed in 3.2.3, meaning I can begin looking to implement this.

**Comment by aaliddell** [comment]
This will be fixed in 4.0.0. The require path will by default have the name_pb removed, with an option to use the old behaviour
